iot/http_example/cloudiot_http_example_nodejs.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const jwt = require('jsonwebtoken');
+const request = require('request');
+
+console.log('Google Cloud IoT Core HTTP example.');
+var argv = require(`yargs`)
+    .options({
+      project_id: {
+        default: process.env.GCLOUD_PROJECT || process.env.GOOGLE_CLOUD_PROJECT,
+        description: 'The Project ID to use. Defaults to the value of the GCLOUD_PROJECT or GOOGLE_CLOUD_PROJECT environment variables.',
+        requiresArg: true,
+        type: 'string'
+      },
+      cloud_region: {
+        default: 'us-central1',
+        description: 'GCP cloud region.',
+        requiresArg: true,
+        type: 'string'
+      },
+      registry_id: {
+        description: 'Cloud IoT registry ID.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      device_id: {
+        description: 'Cloud IoT device ID.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      private_key_file: {
+        description: 'Path to private key file.',
+        requiresArg: true,
+        demandOption: true,
+        type: 'string'
+      },
+      algorithm: {
+        description: 'Encryption algorithm to generate the JWT.',
+        requiresArg: true,
+        demandOption: true,
+        choices: ['RS256', 'ES256'],
```

Comment:
If the target user of this sample is expected to be familiar with JWT, or node already prints allowed choices, then please ignore my comment:

It might help to print the allowed choices in the help/description message.